### PR TITLE
feat: API Dynamic page sizing when query exceeds 30 seconds

### PR DESCRIPTION
**Pull request by another-rex**
Do dynamic page sizing for when the query starts to exceed 30 seconds. 

Turns out past me already thought of this, just needed to uncomment the relevant lines.

### CODE REVIEW

**Comment by another-rex** [comment]
A significant amount of time is spend in converting the bug to response. When running locally, I managed to retrieve 93 entries before the time limit is hit, resulting in a 16mb file. The conversion on those 93 entries took 6 seconds. 

When running in the cloud, I expect to be able to retrieve more than 93 entries, therefore the conversion will also take longer. If the conversion takes longer than 30 seconds, this will still timeout. Requires testing this in staging.

**Review by hogo6002 — APPROVED** [review]

File: gcp/api/server.py [hunk 1]
```diff
@@ -992,7 +998,9 @@ def query_by_generic_helper(results: list, cursor, context: QueryContext,
   query: ndb.Query = base_query.filter(osv.Bug.affected_fuzzy == version)
   it: ndb.QueryIterator = query.iter(start_cursor=context.page_token)
   while (yield it.has_next_async()):
-    if len(results) >= context.total_responses.page_limit():
+    now = datetime.now()
```
Comment (oliverchang) [inline gcp/api/server.py:1001]: nit: is it possible to write some kind of wrapper such that there's less  duplication of  this logic across all the places we iterate?

File: gcp/api/test_server.py [hunk 2]
```diff
@@ -139,7 +139,7 @@ def start_esp(port, backend_port, credential_path, log_path):
         '--non_gcp',
         '--enable_debug',
         '--transcoding_preserve_proto_field_names',
-        '--envoy_connection_buffer_limit_bytes=10485760',
+        '--envoy_connection_buffer_limit_bytes=104857600',
```
Comment (oliverchang) [inline gcp/api/test_server.py:142]: is there some prod config we have to change here too?

Comment (another-rex) [reply gcp/api/test_server.py:142]: Yes, just discovered this when running on staging, we need to update the terraform config as well, I'll make a follow-up PR after testing on staging.

**Pull request merged**